### ISSUE TITLE
ConnectionGroup.dm nullrefs due to portals.

### ISSUE THREAD

**Issue by PJB3005** [d3athrow/vgstation13#16405]
```
[22:39:28] Runtime in ConnectionGroup.dm,220: Cannot execute null.return air().
  proc name: New (/connection_edge/unsimulated/New)
  usr: Pete Zuh (megapear) (/mob/living/carbon/human)
  usr.loc: Carpet (237, 242, 1) (/turf/simulated/floor/carpet)
  src: /connection_edge/unsimulated (/connection_edge/unsimulated)
  call stack:
  /connection_edge/unsimulated (/connection_edge/unsimulated): New(Zone 2059 (/zone), null)
  Air (/datum/subsystem/air): get edge(Zone 2059 (/zone), null)
  /connection (/connection): New(the floor (238,242,1) (/turf/simulated/floor/wood), the reinforced wall (230,239,1) (/turf/simulated/wall/r_wall))
  the portal (/obj/effect/portal): connect atmospheres()
  the portal (/obj/effect/portal): New(the floor (238,242,1) (/turf/simulated/floor/wood), 300)
```

**Comment by Exxion** [comment]
Essentially, ZAS connections assume that if a turf is simulated, it has a zone. This is not true for turfs that block air or contain an object that blocks air.
But what actually *should* happen if you try to connect to a zoneless simulated turf? Should the connection just fail? Should it connect the turf's internal atmosphere? Something else?

**Comment by Shadowmech88** [comment]
Yeah, I saw this while testing the warping claws, and intended to fix it after they were merged to avoid conflicts.
I was just going to make the connection fail to form if one of the turfs was dense.

**Comment by Exxion** [comment]
You should check if it has a `zone` specifically rather than if it's simply dense, as there are several ways a turf can have no `zone` even if it is not dense itself.

**Comment by Shadowmech88** [comment]
Is `zone` a var on the turf itself?

**Comment by PJB3005** [comment]
Yes.

**Comment by Shadowmech88** [comment]
So, portal code already checks for zones, via `SSair.has_valid_zone(turf)`. It makes a two-way connection if both portals have valid zones, and a one-way connection if only one of the portals has a valid zone. It was done this way, I believe, because space doesn't have a zone.
Why would a portal being inside a dense object be different than it being in space, in terms of that portal not having a zone?

**Comment by Exxion** [comment]
Because `connection_edge/unsimulated` explicitly requires that the zoneless turf is `unsimulated`
Otherwise, clever usage of portals and injectors could give you infinite gas

**Comment by Shadowmech88** [comment]
So after checking whether the portal's own turf has a zone, I should check whether the opposite portal's turf has a zone or is `/unsimulated` before making even a one-way connection?

**Comment by Exxion** [comment]
What do you mean one-way connection?

Actually wait
Why do portals make two connections? Connections between zones are always bidirectional, and connections to space require the space tile is the second arg to `New()`. Either way one of the connections is superfluous.

**Comment by Shadowmech88** [comment]
Based on my comments in the initial PR, it seems that airflow pushing and pulling through the portals did not work unless each portal made an atmos connection from its own turf to the other portal's turf.
At the time, I speculated that this was also the reason why airflow pulling to space through portals did not work, since the air master never has a valid zone for space turfs, meaning a portal opened between a regular turf and space will only have a single connection, that being from the regular turf to the space turf. Air still drains to space, but no objects are pulled toward the portal.

**Comment by Exxion** [comment]
And yet airflow pulling/pushing works fine between zones, which have only one connection between them.

**Comment by Shadowmech88** [comment]
I don't know nearly enough about atmos code to diagnose why just one connection doesn't work for portals, unfortunately.

**Comment by Exxion** [comment]
Ah, I see the reason.
It's a limitation with `connection`s that honestly shouldn't exist
I'll deal with it
